client-api/src/main/java/org/sonarsource/sonarlint/core/client/api/util/FileUtils.java

```diff
@@ -152,4 +162,47 @@ public static void replaceDir(Consumer<Path> dirContentUpdater, Path target, Pat
     FileUtils.mkdirs(target.getParent());
     FileUtils.moveDir(work, target);
   }
+  
+  /**
+   * On Windows, retries the provided IO operation a number of times, in an effort to make the operation succeed.
+   * 
+   * Operations that might fail on Windows are file & directory move, as well as file deletion. These failures 
+   * are typically caused by the virus scanner and/or the Windows Indexing Service. These services tend to open a file handle 
+   * on newly created files in an effort to scan their content.
+   * 
+   * @param runnable the runnable whose execution should be retried
+   */
+  private static void retry(IORunnable runnable) throws IOException {
+    for (int retries = isWindows()? 30: 1; retries > 0; retries--) {
+      if (retries > 1) {    	  
+	    try {
+	      runnable.run();
+	      break;
+	    } catch(AccessDeniedException e) {
+	      // Sleep a bit to give a chance to the virus scanner / Windows Indexing Service to release the opened file handle
+	      try {
+	        Thread.sleep(100);
+	      } catch (InterruptedException ie) {
+	    	// Nothing meaningful to be done here
+	      }
+	    }
```

Comment:
Please use indentation consistent with the rest of the file. (2 spaces per level)

Reply:
Sorry for that - missed that one (we still use the old 4 spaces per level in our own codebase).